tests/Tests/QuickCheckUtils.hs

```diff
@@ -250,23 +250,21 @@ write_read :: (NFData a, Eq a, Show a)
            -> [a]
            -> Property
 write_read _ _ _ _ (IO.NewlineMode IO.LF IO.CRLF) _ _ = discard
-write_read unline filt writer reader nl buf ts = ioProperty $
-    (===t) <$> act
+write_read _ _ _ _ _ (IO.BlockBuffering (Just x))  _ | x < 4 = discard
+write_read unline filt writer reader nl buf ts
+  = forAll (elements encodings) $ \enc -> ioProperty $ do
```

Comment:
It's somewhat unfortunately that all encodings will be lumped into the same `Property`. Let's refactor this further: make `write_read` to return not a `Property` but `[TestTree]` with a test per encoding.